geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/index/AttributeIndex.scala

```diff
@@ -269,6 +269,9 @@ object AttributeIndex {
   // store 2 bytes for the index of the attribute in the sft - this allows up to 32k attributes in the sft.
   def indexToBytes(i: Int): Array[Byte] = Array((i << 8).asInstanceOf[Byte], i.asInstanceOf[Byte])
 
+  // TODO verify this
+  def bytesToIndex(b0: Byte, b1: Byte): Short = ((b0 >> 8) + b1).toShort
```

Comment:
the `indexToBytes` method is actually broken for values > 256... I replaced them both with guava `Shorts.to/fromBytes`. Probably anyone indexing more than 256 attributes isn't working right atm.